### DISCUSSION TITLE
how does iperf3 calculate bitrate? Is retransmission data included?

### DISCUSSION THREAD

**Discussion by liujian-930212** [esnet/iperf#1682] [General]
Let's say iperf3 server send 100G bits to client during 60s, and 20G bits are retransmission data, 80G bits are effective data. So the bitrate will be 100Gb/60s or 80Gb/60s

Thanks

**Comment by davidBar-On** [comment]
It will be 80Gb/60s.  iperf3 take into account only the data it sends, i.e. 80G in this example.  Retransmissions are done by the TCP stack and are not visible to iperf3.

**Comment by bmah888** [comment]
Also note that iperf3 only counts payload bytes...overheads such as Ethernet/IP/TCP headers are not included.
